eng/pipelines/templates/jobs/archetype-sdk-data.yml

```diff
@@ -0,0 +1,222 @@
+parameters:
+  SDKType: not-specified # Set a default that breaks in obvious ways.
+  ServiceDirectory: not-specified # Set a default that breaks in obvious ways.
+  PreTestSteps: []
+  TestOptions: '$(DefaultOptions)'
+  TestGoals: verify
+  TestMatrix:
+    Linux - Java 8:
+      OSName: 'Linux'
+      OSVmImage: 'ubuntu-16.04'
+      JavaVersion: '1.8'
+      RunTitle: 'Linux on Java 1.8'
+    macOS - Java 8:
+      OSName: 'macOS'
+      OSVmImage: 'macOS-10.13'
+      JavaVersion: '1.8'
+      RunTitle: 'macOS on Java 1.8'
+    Windows - Java 8:
+      OSName: 'Windows'
+      OSVmImage: 'windows-2019'
+      JavaVersion: '1.8'
+      RunTitle: 'Windows on Java 1.8'
+    Linux - Java 11:
+      OSName: 'Linux'
+      OSVmImage: 'ubuntu-16.04'
+      JavaVersion: '1.11'
+      RunTitle: 'Linux on Java 1.11'
+    macOS - Java 11:
+      OSName: 'macOS'
+      OSVmImage: 'macOS-10.13'
+      JavaVersion: '1.11'
+      RunTitle: 'macOS on Java 1.11'
+    Windows - Java 11:
+      OSName: 'Windows'
+      OSVmImage: 'windows-2019'
+      JavaVersion: '1.11'
+      RunTitle: 'Windows on Java 1.11'
+
+jobs:
+  - job: 'Build'
+
+    variables:
+      - template: ../variables/globals.yml
+
+    pool:
+      vmImage: 'ubuntu-16.04'
+
+    strategy:
+      matrix:
+        Client Libraries:
+          ArtifactName: 'packages'
+          JavaVersion: '1.11'
+
+    steps:
+      - script: |
```

Comment:
Orthogonal but potentially more clean-up. I believe this was added to enabling the publishing of dependency reports, see https://github.com/Azure/azure-sdk-for-java/pull/4238, we should see if that is still necessary after all our refactorings and changes to unified pipelines.

Reply:
@mitchdenny, it looks like the reason this tag step was being done was for the nightly dependency report. It seems that this should be a step within the [nightly aggregate report generation](https://github.com/Azure/azure-sdk-for-java/blob/master/eng/pipelines/aggregate-reports.yml). Though it's unclear if this step is actually still necessary.

Reply:
I haven't touched the dependency reports, I think they still run off one of the tagged builds. I think we should file a separate issue for that because we almost certainly will want to move that over to the aggregate reports.

Reply:
OK @mitchdenny, I'll leave this alone in my PR and let you file the issue.

Reply:
Logged: https://github.com/Azure/azure-sdk-for-java/issues/7592